crates/swc_ecma_transforms_base/tests/resolver/issues/7685/output.js

```diff
@@ -2,4 +2,4 @@ var NaN__2 = 1;
 {
     let NaN__3 = 1;
     console.log(NaN__3);
-}console.log(NaN);
+}console.log(NaN__2);
```

Comment:
The difference between this and #8469 is #8469 would generate
```js
var NaN = 1;
{
    let NaN__3 = 1;
    console.log(NaN__3);
}console.log(NaN);
```
which I believe would introduce problems for following passes.